feature_engine/creation/combine_with_reference_feature.py

```diff
@@ -286,6 +294,12 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         if self.new_variables_names:
             X.columns = original_col_names + self.new_variables_names
 
+        if self.drop_original:
+            X.drop(
+                columns=pd.unique(self.variables_to_combine + self.reference_variables),
```

Comment:
I hear that set() instead of pd.unique() is more performant. As in faster. We should probably change.

Reply:
Oh is it? I thought of that for a second. Alright